### PR TITLE
template configmap values

### PR DESCRIPTION
**Pull request by bkomraz1**
Signed-off-by: Boris Komraz <bkomraz1@gmail.com>

### Description
configmap should be templated 
 
### Issues Resolved
https://github.com/opensearch-project/helm-charts/issues/315
 
### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by bkomraz1** [comment]
Adding @peterzhuamazon

**Review by DandyDeveloper — CHANGES_REQUESTED** [review]
Looks like the yml `tpl` isn't working. 

image

I think in your defaults you meant `.Values.serverHost`? 


Beyond this, everything looks good. Fix the conflicts, I'll approve!

**Comment by bkomraz1** [comment]
> Looks like the yml `tpl` isn't working.
> 
> image
> 
> I think in your defaults you meant `.Values.serverHost`?
> 
> Beyond this, everything looks good. Fix the conflicts, I'll approve!

I fixed values.yaml:

```
  #  opensearch_dashboards.yml:
  #   server:
  #     name: dashboards
  #     host: "{{ .Values.serverHost }}"
```

```
helm template . | grep        host: -C 5
    app.kubernetes.io/version: "2.2.0"
    app.kubernetes.io/managed-by: Helm
data:
  opensearch_dashboards.yml: |
    server:
      host: '0.0.0.0'
      name: dashboards
---

**Review by DandyDeveloper — APPROVED** [review]
Looks good, the only one recommendation I'd have is perhaps moving this to a function in the `_helpers.tpl` but since these are two different charts. This might actually be more effort then its worth.

LGTM regardless!

**Comment by peterzhuamazon** [comment]
Sorry for the delay @bkomraz1 adding @prudhvigodithi to take a look.
Thanks.

File: charts/opensearch-dashboards/values.yaml [hunk 1]
```diff
@@ -71,10 +71,17 @@ securityContext:
...
   #  opensearch_dashboards.yml: |
   #   server:
   #     name: dashboards
-  #     host: 0.0.0.0
+  #     host: "{{ .Values.serverHost }}"
+
+  #  opensearch_dashboards.yml:
+  #   server:
+  #     name: dashboards
+  #     host: "{{ .Values.serverHost }}"
```
Comment (peterzhuamazon) [inline charts/opensearch-dashboards/values.yaml:83]: Is it possible to add these changes in README?

**Comment by prudhvigodithi** [comment]
Hey @bkomraz1 thanks for the PR, the only test I was looking for, does `helm upgrade` with this change breaks for works fine, can you please post your testing here?
Thank you

**Comment by bkomraz1** [comment]
```
(⎈ |docker-desktop:default):opensearch$ 
cd opensearch-project-helm-charts/charts/opensearch
helm search repo opensearch
NAME                            	CHART VERSION	APP VERSION	DESCRIPTION
opensearch/opensearch           	2.6.0        	2.3.0      	A Helm chart for OpenSearch
opensearch/opensearch-dashboards	2.5.1        	2.3.0      	A Helm chart for OpenSearch Dashboards
(⎈ |docker-desktop:default):opensearch$ helm install opensearch opensearch/opensearch
NAME: opensearch
LAST DEPLOYED: Mon Sep 19 23:15:04 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Watch all cluster members come up.
  $ kubectl get pods --namespace=default -l app.kubernetes.io/component=opensearch-cluster-master -w
(⎈ |docker-desktop:default):opensearch$ helm upgrade opensearch .
Release "opensearch" has been upgraded. Happy Helming!
NAME: opensearch
LAST DEPLOYED: Mon Sep 19 23:15:44 2022
NAMESPACE: default
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
Watch all cluster members come up.
  $ kubectl get pods --namespace=default -l app.kubernetes.io/component=opensearch-cluster-master -w
(⎈ |docker-desktop:default):opensearch$ helm list
NAME      	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART           	APP VERSION
opensearch	default  	2       	2022-09-19 23:15:44.002241 +0300 IDT	deployed	opensearch-2.6.1	2.3.0

**Comment by prudhvigodithi** [comment]
Hey @bkomraz1 thanks, I wanted to get a check with your change in place and if helm upgrade works with no issues. In general any chart changes should not break the helm upgrade as well, probably its good to add a CI test @peterzhuamazon @DandyDeveloper @TheAlgo @bbarani thoughts ?
Thank you

**Comment by prudhvigodithi** [comment]
Hey @bkomraz1 I now see closely you have done upgrade from the local folder with your changes.

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
@bkomraz1 can you please backport the change to `1.x` branch as well?

**Comment by prudhvigodithi** [comment]
I have created a backport PR https://github.com/opensearch-project/helm-charts/pull/331

**Pull request merged**